relay-server/src/utils/dynamic_sampling.rs

```diff
@@ -119,20 +76,26 @@ pub fn is_trace_fully_sampled(
         return Some(false);
     }
 
-    let sampling_result = get_sampling_result(
-        processing_enabled,
-        None,
-        Some(root_project_state),
-        Some(dsc),
-        None,
-    );
-
-    let sampled = match sampling_result {
-        SamplingResult::Keep => true,
-        SamplingResult::Drop(_) => false,
+    if root_project_config.unsupported() {
+        if processing_enabled {
+            relay_log::error!("found unsupported rules even as processing relay");
+        } else {
+            return Some(true);
+        }
+    }
```

Comment:
since we no longer check unsupported rules in sampling config, we have to do it here instead, this code replaces: https://github.com/getsentry/relay/blob/4d0650ee8a2cfa70adaef2a51774f266857ab367/relay-sampling/src/evaluation.rs#L54

here it returns Some(true) because before this PR, it would return an Err, which would return `Option<SamplingMatch>::None` in `fn merge_configs_and_match`, which would turn into a `SamplingResult::Keep`, which would turn into a `true` in this function.